src/backend/renderer/utils.rs

```diff
@@ -191,7 +191,19 @@ impl RendererSurfaceState {
         }
     }
 
-    pub(crate) fn damage_since(&self, commit: Option<usize>) -> Vec<Rectangle<i32, BufferCoord>> {
+    /// Get the current commit position of this surface
+    ///
+    /// The position should be saved after calling [`damage_since`] and
+    /// provided as the commit in the next call.
```

Comment:
Sounds like this should be returned with damage_since: `-> (Vec<Rectangle<i32, BufferCoord>>, usize)

Reply:
Yeah, good point, this is more an atomic operation so it seems like a good idea to return both